spec/models/contact_spec.rb

```diff
@@ -64,6 +64,20 @@
     end
   end
 
+  describe "contact_type_id" do
+    it "prevents contact_type_id being updated" do
+      contact = described_class.new(name: "HMP halifax",
+                                    address_line_1: "123 test road",
+                                    address_line_2: "little heath",
+                                    town: "bakersville",
+                                    county: "Mercia",
+                                    postcode: "FE2 9JK",
+                                    data_request_emails: "test@test.com\ntest1@test.com",
+                                    contact_type_id: 1)
+      expect { contact.update_attribute(:contact_type_id, 2) }.to raise_error(ActiveRecord::ActiveRecordError, "contact_type_id is marked as readonly") # rubocop:disable Rails/SkipsModelValidations
```

Comment:
Wasn't quite sure how to test this or whether using "attr_readonly" was the best approach. 

Using 'update!' raised an error saying "contact_type cant be blank". Tried to implement the CategoryReference model but was unclear to me whether that was the best way to do it